### PR TITLE
Fix branch pruning resolution of non boolean conditions

### PR DESCRIPTION
**Pull request by frantzmiccoli**
This is:

```
- [x] a bugfix
- [ ] a new feature
```

Checklist:

- [x] Changes are covered by unit tests
- [x] Code style is respected
- [x] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [x] CHANGELOG.md contains a short summary of the change
- [x] Documentation is updated as necessary

### Why this change is needed?

Branch pruning was not properly handling non boolean conditions.

### CODE REVIEW

File: src/PhpSpreadsheet/Calculation/Calculation.php [hunk 1]
```diff
@@ -3804,13 +3804,19 @@ private function processTokenStack($tokens, $cellID = null, Cell $pCell = null)
...
                     $wrappedItem = end($storeValue);
                     $storeValue = end($wrappedItem);
                 }
 
-                if (isset($storeValue) && (($storeValue !== true)
-                    || ($storeValue === 'Pruned branch'))
+                if (isset($storeValue)
+                    && (
+                        !$storeValueAsBool
+                        || ($storeValue === '#N/A')
```
[the remark this answers is not in the document]
Comment (frantzmiccoli) [reply src/PhpSpreadsheet/Calculation/Calculation.php]: Good remark. @PowerKiKi ? Is this the strategy that has been followed elsewhere? Should we proceed on that?

[the remark this answers is not in the document]
Comment (MarkBaker) [reply src/PhpSpreadsheet/Calculation/Calculation.php]: `Functions::isError($cellValue)` tests for the listed errors, plus the slightly more esoteric `#GETTING_DATA`

[the remark this answers is not in the document]
Comment (frantzmiccoli) [reply src/PhpSpreadsheet/Calculation/Calculation.php]: Will modify this accordingly

**Comment by frantzmiccoli** [comment]
Modified accordingly

**Comment by rolandsusans** [comment]
LGTM, good job 👍

**Comment by PowerKiKi** [comment]
Thank you for following up on this work !

**Pull request closed without merging**